### DISCUSSION TITLE
Custom WebhookCall collides with \Spatie\WebhookClient\Jobs\ProcessWebhookJob

### DISCUSSION THREAD

**Discussion by cord** [spatie/laravel-webhook-client#135] [Q&A]
I have implemented a customer `myWebhookCall` model and a job, works so far. 

How can I pass my custom Model into the job?

I tried this:

```
namespace App\Jobs;

use Spatie\WebhookClient\Jobs\ProcessWebhookJob as SpatieProcessWebhookJob;

class ProcessWebhookJob extends SpatieProcessWebhookJob
{

   public function __construct(
        public myWebhookCall $webhookCall
    )
    {
    }

    public function handle()
    {
       // doing something with custom WebhookCall
    }
}
````

and get

`Type of App\Jobs\ProcessWebhookJob::$webhookCall must be Spatie\WebhookClient\Models\WebhookCall (as in class Spatie\WebhookClient\Jobs\ProcessWebhookJob)`

**Comment by freekmurze** [comment]
Does `myWebhookCall` extends our `Spatie\WebhookClient\Models\WebhookCall`?

**Reply by cord** [reply]
Yes

**Reply by freekmurze** [reply]
Did you set your model here? https://github.com/spatie/laravel-webhook-client/blob/main/config/webhook-client.php#L44
(and have cleared config cache?)

**Reply by cord** [reply]
After going through the code in `\Spatie\WebhookClient\WebhookProcessor::processWebhook` I realize my conceptual misunderstanding of how to use the ProcessWebhookJob - for testing I called that job directly which will then misses the configuration context.

Thanks again for your great work and support!

**Reply by janakaedirisinghe** [reply]
@cord can you please explain this ?? how did you resolve this with your custom model?

**Reply by michaelklopf** [reply]
@janakaedirisinghe 
He probably didn't use the `webhook-client.php` to configure his webhook and dispatched the job himself? So now he uses the config and added his custom model over there.

**Reply by janakaedirisinghe** [reply]
thanks @michaelklopf  actually Im working with the spatie multitenancy package and I have managed to add custom model and get the webhook response overit.. thank you.
